Order/Order.Application/Services/ConsumerService.cs

```diff
@@ -0,0 +1,88 @@
+using AutoMapper;
+using Confluent.Kafka;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Order.Application.Constants;
+using Order.Application.DTOs.UserDTOs;
+using Order.Application.Exceptions;
+using Order.Application.Extensions;
+using Order.Application.Interfaces;
+using Order.Domain.Entities;
+using Order.Domain.Interfaces;
+
+namespace Order.Application.Services;
+
+public class ConsumerService: BackgroundService 
+{
+	private readonly IConsumer<long, UserMessageDto> _consumer;	
+	private readonly IServiceProvider _serviceProvider;
+	
+	public ConsumerService(IConsumer<long, UserMessageDto> consumer, IServiceProvider provider)
+	{
+		_consumer = consumer;
+		_serviceProvider = provider;
+	}
+
+	protected override Task ExecuteAsync(CancellationToken stoppingToken)
+	{
+		var topics = new string[]
+		{
+			Topics.UserCreated,
+			Topics.UserDeleted,
+			Topics.UserUpdated
+		};
+		
+		_consumer.Subscribe(topics);
+		
+		return Task.Run(async () =>
```

Comment:
why do you decide to use ask.Run instead of async/await here?

Reply:
Without using Task.Run app stops at _consumer.Consume() and waits for message. I have not found async method of consumption and used Task.Run